drivers: kscan_touch: enable the touch panel for rt1170_evk

CristXu opened the pull request:
Enable the touch panel for the rt1170_evk

Signed-off-by: Crist Xu <crist.xu@nxp.com>

CristXu committed:
drivers: kscan_touch: enable the touch panel for rt1170

Enable the touch panel for the rt1170_evk

Signed-off-by: Crist Xu <crist.xu@nxp.com>

galak commented on dts/arm/nxp/nxp_rt1170_cm7.dtsi:
```diff
@@ -48,7 +48,8 @@
 			reg = <0x42008000 0x4000>;
-			interrupts = <102 0>, <103 0>;
+			interrupts = <99 0>;
+			label = "GPIO_2";
```
remove `label` prop

galak replied:
This change for IRQs should be in a separate commit.

dleach02 replied:
@galak, when do you think it is time to add a CI check for the label prop?

CristXu replied:
Will change in another commi

galak commented on dts/bindings/kscan/goodix,gt911.yaml:
```diff
@@ -0,0 +1,16 @@
+include: [kscan.yaml, i2c-device.yaml]
+
+properties:
+    int-gpios:
```
rename to `irq-gpios` to match linux binding.

CristXu replied:
done

galak commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:
```diff
@@ -101,8 +102,18 @@
+	touch_controller: gt911@5d {
+		compatible = "goodix,gt911";
+		reg = <0x5d>;
+		label = "GT911";
```
remove `label` prop.

CristXu replied:
done

galak commented on boards/arm/mimxrt1170_evk/Kconfig.defconfig:
```diff
@@ -33,6 +33,19 @@ endchoice
 
 endif #FLASH
 
+config I2C
```
remove `config I2C` bits, those should get dealt with by the Kconfig of the driver itself.

CristXu replied:
done

galak commented on boards/arm/mimxrt1170_evk/Kconfig.defconfig:
```diff
@@ -33,6 +33,19 @@ endchoice
+
+if KSCAN
+
+config KSCAN_GT911
```
remove `KSCAN_GT911`, will get enabled by Kconfig and devicetree association.

CristXu replied:
Done

galak commented on drivers/kscan/Kconfig.gt911:
```diff
@@ -0,0 +1,27 @@
+
+menuconfig KSCAN_GT911
+	bool "GT9xx / GT9xxx capacitive touch panel driver"
+	depends on I2C
```
replace `depends on I2C` with:

```
default y
depends on DT_HAS_GOODIX_GT911_ENABLED
select I2C
```

CristXu replied:
Done

galak reviewed (changes requested):
A few Kconfig and dts related changes.

CristXu commented on dts/bindings/kscan/goodix,gt911.yaml:
```diff
@@ -0,0 +1,16 @@
+    int-gpios:
+      type: phandle-array
+      required: false
+    rst-gpios:
```
done

CristXu committed:
dts: rt1170: Change the interrupt idx of the fast-gpio2

The fast-gpio has a dedicated interrupt idx 99, not
the same with the normal gpio. So if we want to use
the fast-gpio, need use the interrupt idx 99.

Signed-off-by: Crist Xu <crist.xu@nxp.com>

galak commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:
```diff
@@ -15,6 +15,7 @@
 
 	aliases {
 		mipi-dsi = &mipi_dsi;
+		kscan0 = &touch_controller;
```
replace alias `kscan0`, with a chosen node property:

`zephyr,keyboard-scan = &touch_controller;`

CristXu replied:
done

galak reviewed (dismissed):
minor change to dtsi

mbolivar-nordic commented on boards/arm/mimxrt1170_evk/mimxrt1170_evk_cm7.dts:
```diff
@@ -29,6 +29,7 @@
 		zephyr,flash = &is25wp128;
 		zephyr,code-partition = &slot0_partition;
 		zephyr,display = &lcdif;
+		zephyr,keyboard-scan = &touch_controller;
```
@CristXu you need to update the table at the bottom of doc/build/dts/api.rst if you're going to add a new chosen node

@galak don't forget to mention this too please

galak replied:
@mbolivar-nordic its in that table already - https://docs.zephyrproject.org/latest/build/dts/api/api.html#id2

mbolivar-nordic reviewed (approved):
whoops thanks @galak
